core/lib/mno_enterprise/concerns/controllers/auth/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,209 @@
+
+          @user = MnoEnterprise::User.find_for_oauth(auth, opts, current_user)
+
+          if @user.persisted?
+            # # TODO: not working in Maestrano, reimplement
+            # if app_id = env['omniauth.params']['bundle']
+            #   apps = MnoEnterprise::App.where(id: app_id).pluck(:nid)
+            #   app_instances = setup_apps(@user, apps)
+            # end
+
+            sign_in_and_redirect @user, event: :authentication
+            set_flash_message(:notice, :success, kind: "#{provider}".capitalize) if is_navigational_format?
+          else
+            session["devise.#{provider}_data"] = env["omniauth.auth"]
+            redirect_to new_user_registration_url
+          end
+        end
+      }
+    end
+  end
+
+  #==================================================================
+  # Instance methods
+  #==================================================================
+  # GET|POST /users/auth/:action/callback
+  def intuit
+    auth = request.env['omniauth.auth']
+    opts = {
+      orga_on_create: create_orga_on_user_creation(auth.info.email),
+      authorized_link_to_email: session['omniauth.intuit.authorized_link_to_email']
+    }
+
+    # Try to find via intuit
+    begin
+      @user = MnoEnterprise::User.find_for_oauth(auth, opts, current_user)
+    rescue SecurityError
+      # Intuit email is NOT a confirmed email. Therefore we need to ask the user to
+      # login the old fashion to make sure it is the right user!
+      session["omniauth.intuit.request_account_link"] = true
+      redirect_to new_user_session_path, notice: "Please sign in using your regular Maestrano account to confirm that you want to link it to your Intuit account"
+      return
+    end
+
+    # Cleanup any temporary omniauth.intuit session
+    cleanup_intuit_session
+
+    if @user && @user.persisted?
+      # Automatically adds a QuickBooks app (and any other app passed via :app param)
+      # to the user orga
+      # Only for new users for which an orga was created (not an invited user
+      # typically)
+      app_instances = setup_apps(@user,['quickbooks',params[:app]], oauth_keyset: params[:app])
+      qb_instance = app_instances.first
+
+      # On Intuit, Mno is configured to add qb_initiated=true if the user
+      # comes directly from apps.com (This is a different workflow from using
+      # the QuickBooks connect button because we're supposed to trigger the
+      # oauth workflow directly via javascript using directConnectToIntuit)
+      # Here we store in session the fact that we need to trigger an oauth
+      # workflow via directConnectToIntuit
+      # ----
+      # See layouts/partners/intuit for more info. The session param set
+      # below get reset in the view.
+      #
+      if params[:qb_initiated] && qb_instance && !qb_instance.oauth_keys_valid?
+        session[:qb_direct_connect_grant_url] = authorize_webhook_oauth_url(qb_instance.uid)
```

Comment:
Actually this may not be required anymore (unless we have some JS in the frontend to support this)